--- a/dsdt.dsl
+++ b/dsdt.dsl
@@ -18,7 +18,7 @@
  *     Compiler ID      "INTL"
  *     Compiler Version 0x20160527 (538314023)
  */
-DefinitionBlock ("", "DSDT", 2, "LENOVO", "SKL     ", 0x00000000)
+DefinitionBlock ("", "DSDT", 2, "LENOVO", "SKL     ", 0x00000001)
 {
     External (_PR_.BGIA, UnknownObj)
     External (_PR_.BGMA, UnknownObj)
@@ -232,9 +232,7 @@
     Name (SS1, 0x00)
     Name (SS2, 0x00)
     Name (SS3, One)
-    One
     Name (SS4, One)
-    One
     OperationRegion (GNVS, SystemMemory, 0x4FF34000, 0x0789)
     Field (GNVS, AnyAcc, Lock, Preserve)
     {
@@ -13798,8 +13796,7 @@
                     If ((DerefOf (Arg0 [0x00]) == 0x01))
                     {
                         ADBG (Concatenate ("POFF GPIO=", ToHexString (DerefOf (Arg0 [0x02]))))
-                        \_SB.SGOV (DerefOf (Arg0 [0x02]), (DerefOf (Arg0 [0x03]) ^ 
-                            0x01))
+                        \_SB.SGOV (DerefOf (Arg0 [0x02]), (DerefOf (Arg0 [0x03]) ^ 0x01))
                     }
 
                     If ((DerefOf (Arg0 [0x00]) == 0x02))
@@ -30797,6 +30794,13 @@
         0x00, 
         0x00
     })
+    Name (\_S3, Package (0x04)  // _S3_: S3 System State
+    {
+        0x05,
+        0x05,
+        0x00,
+        0x00
+    })
     Name (\_S4, Package (0x04)  // _S4_: S4 System State
     {
         0x06, 
@@ -37011,6 +37015,11 @@
                 {
                     Return (0x0D)
                 }
+
+                Method (HEBC, 0, Serialized)
+                {
+                    Return (0x340000)
+                }
             }
         }
 
